### DISCUSSION TITLE
Connect a username to an email | Who is who?

### DISCUSSION THREAD

**Discussion by ggsibarite** [community/community#31611] [Other Feature Feedback, Questions, & Ideas]
Hello everyone,
I created a repository and I gave access to it to about 10 persons. Now, they all created a Github profile to access but on the page to manage Members, all I can see is a list of usernames and I cannot relate it easily to each physical person. Any possibility to check for each user who has access to my repository his/her email address?
Thanks for your help !

**Comment by jfullstackdev** [comment]
I'm wondering why you gave the access without knowing exactly their GitHub profile, because you will be doing it manually. They can't have access if you did not give them in the first place.

**Reply by ggsibarite** [reply]
I think I didn't explain it correctly. Consider I send invitations to (fake names..) : francesco.bianchi@gmail.com AND francesca.tutti@gmail.com.
As they don't have an existing Github profile, they create one using the invitation they got to join my repository.
Once they both accept my invitation, I cannot see their email address anymore but instead I see their GitHub profile with their username only.
Then, how can I distinguish using their username (let's say Francesc123 AND Francesc456) who is who?

**Comment by airtower-luna** [comment]
If they use the email addresses you know as author email for their commits, you should be able to make the connection from any commit by them:

* Look at the commit on GitHub, and you can see which account GitHub connects the commit to.
* Check the author email, either using `git log`, or by appending `.patch` to the commit URL (that returns the patch in email format, with the author info in the `From` header).

GitHub uses the author email to connect commits to accounts, so this will tell you that the account belongs to someone with that email address (roughly, because unverified addresses work for commit association, too).

**Comment by ggsibarite** [comment]
Thank you for your answer, I had thought about it already (email related to commit).
However, I mostly use the repository to share data with people (professors) so they are almost no chance that they will commit anything, their only use of Github is to get access to what I am sharing. Among the 10 people that have access to the repository, maybe just 2 or 3 (students) will commit anything besides me. The rest just need to get a viewing access.

**Comment by jfullstackdev** [comment]
seems you know them, why not simply ask them who is who
